File: puzzles/python3/mime_type.py
from typing import Dict


if __name__ == "__main__":
    table: Dict[str, str] = {}  # file extension => mime type
    nb_elements: int = int(input())
    nb_names: int = int(input())

    for _ in range(nb_elements):
        extension, mime_type = input().split()
        table[extension.lower()] = mime_type

    for _ in range(nb_names):
        name: str = input().lower()
        dot_index: int = name.rfind(".")
        if dot_index == -1 or dot_index == len(name) - 1:
            print("UNKNOWN")
        else:
            extension: str = name[dot_index + 1:]
            if extension in table:
                print(table[extension])
            else:
                print("UNKNOWN")

File: puzzles/go/mime-type.go
package main

import (
  "fmt"
  "os"
  "bufio"
  "strings"
)

func main() {
  scanner := bufio.NewScanner(os.Stdin)
  scanner.Buffer(make([]byte, 1000000), 1000000)

  // dictionary to map file extensions to mime types
  var table = make(map[string]string)
  var nbElements int
  scanner.Scan()
  fmt.Sscan(scanner.Text(), &nbElements)
  
  var nbNames int
  scanner.Scan()
  fmt.Sscan(scanner.Text(), &nbNames)
  
  for i := 0; i < nbElements; i++ {
    var extension, mimeType string
    scanner.Scan()
    fmt.Sscan(scanner.Text(), &extension, &mimeType)
    table[strings.ToLower(extension)] = mimeType
  }

  for i := 0; i < nbNames; i++ {
    scanner.Scan()
    name := strings.ToLower(scanner.Text())
    var dotIndex = strings.LastIndex(name, ".")
    if dotIndex == -1 || dotIndex == len(name) - 1 {
      fmt.Println("UNKNOWN")
    } else {
      var extension = name[dotIndex + 1:]
      mimeType, ok := table[extension]
      if ok {
        fmt.Println(mimeType)
      } else {
        fmt.Println("UNKNOWN")
      }
    }
  }
}
